dbt_platform_helper/providers/opensearch.py

```diff
@@ -0,0 +1,36 @@
+from dbt_platform_helper.providers.cache import CacheProvider
+
+
+class OpensearchProvider:
+
+    def __init__(self, opensearch_client):
+        self.openserarch_client = opensearch_client
+
+    def get_supported_opensearch_versions(self):
+
+        cache_provider = self.__get_cache_provider()
+
+        if cache_provider.cache_refresh_required("opensearch"):
+
+            response = self.openserarch_client.list_versions()
+            all_versions = response["Versions"]
+
+            opensearch_versions = [
+                version for version in all_versions if not version.startswith("Elasticsearch_")
+            ]
+            supported_versions = [
+                version.removeprefix("OpenSearch_") for version in opensearch_versions
+            ]
+
+            cache_provider.update_cache("opensearch", supported_versions)
+
+            return supported_versions
+
+        else:
+            return cache_provider.read_supported_versions_from_cache("opensearch")
+
+    # TODO - cache provider instantiated here rather than via dependancy injection since it will likely only be used in the get_supported_redis_versions method.
```

Comment:
I guess this comment needs to be updated to `get_supported_opensearch_versions` function